### ISSUE TITLE
numa_hwloc.cc:121 Call to hwloc_set_cpubind() failed: Invalid argument

### ISSUE THREAD

**Issue by carlosgmartin** [openxla/xla#45484]
**System information:**

- uname: Linux 4.18.0-372.32.1.el8_6.x86_64
- OS name: Springdale Open Enterprise Linux
- OS version: 8.6 (Modena)
- Python version: 3.12.13
- JAX version: 0.10.2
- GPU model and memory: NVIDIA A100-SXM4-40GB

**Problem description:**

On a SLURM cluster, when initializing XLA, it spams the console with `ERROR` level logs. These originate from line 121 of `numa_hwloc.cc`.

Example:

```shell
$ srun --partition=gpu --gpus=1 python3 -c 'import jax; print(jax.devices())'
E0710 03:21:27.670090 2522223 numa_hwloc.cc:121] Call to hwloc_set_cpubind() failed: Invalid argument [22]
[CudaDevice(id=0)]
```

Because the cluster uses standard cgroups to restrict which CPU cores a SLURM job can access, XLA's attempt to bind threads via `hwloc_set_cpubind()` is rejected by the OS with `EINVAL` (Invalid argument [22]).

Since XLA gracefully falls back to the default OS thread scheduler and GPU execution continues normally, logging this as a fatal-looking `ERROR` is misleading.

**Proposed solution:**

Consider downgrading this specific `hwloc` binding failure from `LOG(ERROR)` to `VLOG(1)` or `LOG(WARNING)`. Alternatively, XLA could silently catch `EINVAL` during thread pinning, given that failing to pin a thread is a standard, non-fatal operational reality on modern HPC clusters.
